File: frontend/src/interfaces/Auction.ts
export enum AuctionStatus {
    OPEN = 'OPEN',
    PENDING = 'PENDING',
    CLOSED = 'CLOSED',
}

export interface Auction {
    id: number;
    startPrice?: number | null;
    startTime?: string | null;
    endTime?: string | null;
    status: AuctionStatus;
    deliverySlotId?: number | null;
    slotStartTime?: string | null;
    slotEndTime?: string | null;
    whaleOnly?: boolean;
}


export type CreateAuction = {
    startPrice: string,
    startTime: string, // ISO FORMAT
    "endTime":string, // ISO FORMAT
    "status": AuctionStatus,
    "deliverySlotId": number,
    "whaleOnly"?: boolean
}

File: backend/src/main/java/com/delivery/auction/entity/Auction.java
package com.delivery.auction.entity;

import com.delivery.deliverySlot.entity.DeliverySlot;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "auctions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Auction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Float startPrice;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    @Enumerated(EnumType.STRING)
    private AuctionStatus status;

    @ManyToOne
    @JoinColumn(name = "delivery_slot_id")
    private DeliverySlot deliverySlot;

    private Boolean whaleOnly = false;
}
